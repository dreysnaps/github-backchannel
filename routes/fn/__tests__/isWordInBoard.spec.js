const { isWordInBoard } = require("../isWordInBoard");

const DEFAULT_BOARD = "T, A, P, *, E, A, K, S, O, B, R, S, S, *, X, D";

describe("Prep Data", () => {
  it("Wait for Dictionary to be populated.", done => {
    setTimeout(() => {
      expect(true).toBe(true);
      done();
    }, 1000);
  });
});

describe("Check Invalid Words", () => {
  it("Check words which do not exist.", async () => {
    expect(await isWordInBoard(DEFAULT_BOARD, "ahh")).toMatchSnapshot();
    expect(await isWordInBoard(DEFAULT_BOARD, "ha41")).toMatchSnapshot();
  });
  it("Check words whic exists but are too large", async () => {
    expect(
      await isWordInBoard(DEFAULT_BOARD, "authoritarianism")
    ).toMatchSnapshot();
    expect(
      await isWordInBoard(DEFAULT_BOARD, "counterbalancing")
    ).toMatchSnapshot();
  });
});

describe("Check Valid Words", () => {
  it("Check small letter words", async () => {
    expect(await isWordInBoard(DEFAULT_BOARD, "tap")).toMatchSnapshot();
    expect(await isWordInBoard(DEFAULT_BOARD, "eat")).toMatchSnapshot();
  });

  it("Check capitalized words", async () => {
    expect(await isWordInBoard(DEFAULT_BOARD, "TAP")).toMatchSnapshot();
    expect(await isWordInBoard(DEFAULT_BOARD, "eat")).toMatchSnapshot();
  });

  it("Check words with *", async () => {
    expect(await isWordInBoard(DEFAULT_BOARD, "SOX")).toMatchSnapshot();
    expect(await isWordInBoard(DEFAULT_BOARD, "SIX")).toMatchSnapshot();
    expect(await isWordInBoard(DEFAULT_BOARD, "ASS")).toMatchSnapshot();
  });
});
