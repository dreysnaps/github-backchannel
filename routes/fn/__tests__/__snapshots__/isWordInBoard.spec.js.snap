// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check Invalid Words Check words whic exists but are too large 1`] = `false`;

exports[`Check Invalid Words Check words whic exists but are too large 2`] = `false`;

exports[`Check Invalid Words Check words which do not exist. 1`] = `false`;

exports[`Check Invalid Words Check words which do not exist. 2`] = `false`;

exports[`Check Valid Words Check capitalized words 1`] = `true`;

exports[`Check Valid Words Check capitalized words 2`] = `true`;

exports[`Check Valid Words Check small letter words 1`] = `true`;

exports[`Check Valid Words Check small letter words 2`] = `true`;

exports[`Check Valid Words Check words with * 1`] = `true`;

exports[`Check Valid Words Check words with * 2`] = `true`;

exports[`Check Valid Words Check words with * 3`] = `true`;
